File: Problems/424_Longest_Repeating_Character_Replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        table = [0]*26
        currMaxCnt = 0
        maxLen = 0
        l = 0
        for r in range(len(s)):
            table[ord(s[r])-ord('A')] += 1
            currMaxCnt = max(currMaxCnt, table[ord(s[r])-ord('A')])
            while r-l+1-currMaxCnt>k:
                table[ord(s[l])-ord('A')]-= 1
                l += 1
            maxLen = max(maxLen, r-l+1)
        return maxLen                

File: Problems/424_Longest_Repeating_Character_Replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> table(26, 0);
        int maxCount = 0;
        int maxLen = 0;
        int l = 0;
        for(int r=0; r<s.length(); r++){
            table[s[r]-'A']++;
            maxCount = max(maxCount, table[s[r]-'A']);
            while(r-l+1-maxCount>k){
                table[s[l]-'A']--;
                l++;
            }
            maxLen = max(maxLen, r-l+1);
        }
        return maxLen;
    }
};